### ISSUE TITLE
Blocking specific mastodon users

### ISSUE THREAD

**Issue by bekoli** [snarfed/bridgy#895]
Same problem like #889 with Mastodon. Talking to my instance admin to block some mindless bot accounts instance wide. That won't help with all cases, of course. Reading the block list from the instance is probably the way to go. I dunno about the API but the block list can be exported as CSV via e.g. https://social.tchncs.de/settings/exports/blocks.csv as well.

I read https://indieweb.org/moderation but processing spam that occurs _on each single post made_ doesn't sounds feasible as well.

Mastodon has a `bot` flag btw. Not all bots abide, of course, but it may be an idea to ignore such reactions at all as well.

**Comment by snarfed** [comment]
thanks for the suggestion! yup, i've been considering using your per-user Mastodon block list, like we already do for Twitter: https://brid.gy/about#blocked .

the catch is the domain (ie instance) level blocks, https://docs.joinmastodon.org/usage/privacy/#hiding-an-entire-server , which Twitter doesn't have. this is tracked in snarfed/granary#173. it's not hard to implement, i just need a way to represent it in ActivityStreams 1, granary's base data format.

and yes, we definitely also encourage people to moderate and block within their own sites! you can do anything you want there, even beyond moderation, eg block users and whole domains if you want.

**Comment by snarfed** [comment]
oh, also, if your admin blocks accounts and domains instance wide, i assume your instance won't accept their toots or interactions at all, so i'd never get them from the API in the first place, right?

**Comment by bekoli** [comment]
That's true. If it's blocked instance wide nothing will show up from my understanding as well.
